optimum/habana/transformers/trainer.py

```diff
@@ -853,7 +853,7 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
                     self.control = self.callback_handler.on_step_begin(args, self.state, self.control)
 
                 # attn_softmax_bf16 is enabled only for llama
-                if hasattr(self.model, "generation_config"):
+                if hasattr(self.model, "generation_config") and self.model.generation_config is not None:
```

Comment:
Not sure we need to add this. If `self.model.generation_config.attn_softmax_bf16` is None, the condition `self.model.generation_config.attn_softmax_bf16` in the if-statement right below will not be True and thus nothing is done no?

Reply:
We met an error saying:  'NoneType' object has no attribute 'attn_softmax_bf16' so had to check if self.model.generation_config is not None

Reply:
Ah my bad I read the code too quickly!
How did you get that error? What was the command or use case?

Reply:
We got the error why FT llama2-7b

Reply:
for sequence classification